### PR TITLE
Add eitem_type field to EItem and 'MULTIMEDIA' to DOCUMENT_TYPES

### LINKED ISSUE #805: New Audiobooks and Videos document types
The O'Reilly (SAFARI) platform also provides audiobooks and videos.

### New Content Types
AUDIOBOOK
VIDEO

When importing, distinguish:
BOOK -> am in field LDR
AUDIOBOOK -> im in field LDR
VIDEO -> gm in field LDR

Accordingly, adapt the description on the e-items:
BOOK -> e-book
AUDIOBOOK -> audiobook
VIDEO -> video

In other catalogues, audiobooks and videos are treated as the following:

### Worldcat

*Format*:

* Book
* Print Book
* eBook
* Audiobook
* Video

### Swisscovery

*Resource Type*:

* Films
* Books
* Other (videos under this resource type)
* Audio

*Publication Format*:

* Electronic Books
* Audiobooks
* Electronic Videos
* Online Resource
* Multimedias

### TIB

*Type of media*:
* Book
* Audio-visual
* Audio file

*Type of material*:
* Electronic resources
* Print
* Audio-visual

### British Library

*Material Type*:
* Moving Image
* Books
* Audio

*Access Options*:
* Online
* Shelved

### Zenodo

*Type*:
* Publication
* Book
* Video/Audio

### Amazon
*See all formats and editions*:
* Kindle
* Audiobook
* Hardcover
* Paperback

## Proposal 
- An Audiobook can be of Document Type Book. Audiobook are added to existing Book record as additional e-item with description 'Audiobook'. If there is no record, a Book record is created with an e-item with description 'Audiobook'. ISBNs identified as '(electronic audio bk.)' (020 subfield q) should refer to Material 'Audiobook'. The other ISBNs should be treated as usual.
- Keep Video separated as they are courses or lessons -> Add a Document Type for Video. Add an e-item with description ‘Video’. Keep only ISBNs identified as '(electronic video)' (020 subfield q), should refer to Material 'Video'.
- Display Audiobook and Video in the Formats in the facets (based on the description in the e-items): e-book/ebook (to be standardized - please bulk update to 'e-book') -> Facet 'E-book'; Audiobook -> Facet 'Audiobook'; Video -> Facet 'Video', in the Filter by Format.
- For Deletions, distinguish Ebook/Audiobook with the LDR field (am (Book)/im (Audiobook)) to remove only the corresponding e-item. The rest of deletion process would as usual.

See attached screenshot:
Screenshot 2023-01-12 at 16 27 34

XML imports:
[xml-example.zip](https://github.com/CERNDocumentServer/cds-ils/files/10403162/xml-example.zip)

### LINKED ISSUE #865: Importer: import audiobooks
- An Audiobook can be of Document Type Book. Audiobook are added to existing Book record as additional e-item with description 'Audiobook'. If there is no record, a Book record is created with an e-item with description 'Audiobook'. ISBNs identified as '(electronic audio bk.)' (020 subfield q) should refer to Material 'Audiobook'. The other ISBNs should be treated as usual. Important: `<leader>` field should include value `im` to identify as audiobook
- Display Audiobook  in the Formats in the facets (based on the description in the e-items): e-book/ebook (to be standardized - please bulk update to 'e-book') -> Facet 'E-book'; Audiobook -> Facet 'Audiobook';  in the Filter by Format.
- For Deletions, distinguish Ebook/Audiobook with the LDR field (am (Book)/im (Audiobook)) to remove only the corresponding e-item. The rest of deletion process would as usual.

### PR DESCRIPTION
**Pull request by sakshamarora1**
:heart: Thank you for your contribution!

### Description

closes: https://github.com/CERNDocumentServer/cds-ils/issues/805
closes: https://github.com/CERNDocumentServer/cds-ils/issues/865

### CODE REVIEW

File: invenio_app_ils/config.py [hunk 1]
```diff
@@ -829,6 +829,7 @@ def _(x):
     eitems=dict(
         aggs=dict(
             access=dict(terms=dict(field="open_access")),
+            eitemtype=dict(terms=dict(field="eitem_type")),
```
Comment (kpsherva) [inline invenio_app_ils/config.py]: for naming consistency: 
```suggestion
            eitem_type=dict(terms=dict(field="eitem_type")),
```

File: invenio_app_ils/config.py [hunk 2]
```diff
@@ -841,6 +842,7 @@ def _(x):
             ),
         ),
         post_filters=dict(
+            eitemtype=terms_filter("eitem_type"),
```
Comment (kpsherva) [inline invenio_app_ils/config.py]:
```suggestion
            eitem_type=terms_filter("eitem_type"),
```

File: invenio_app_ils/eitems/api.py [hunk 3]
```diff
@@ -57,6 +57,12 @@ def validate(self, record, **kwargs):
 class EItem(IlsRecord):
     """EItem record class."""
 
+    EITEM_TYPES = [
```
Comment (kpsherva) [inline invenio_app_ils/eitems/api.py:60]: not a strong opinion, I would like to also check with the others (could you discuss on a standup?): would it be more flexible to store these types in a vocabulary?

Comment (sakshamarora1) [reply invenio_app_ils/eitems/api.py:60]: Sure, I will discuss
But wouldn't keeping the type of the record in its class provide better visibility and readability and also be easier to maintain
if we make it a vocabulary, then we would also need do the same for all other record types, and that will require checking if there might be some other flow that'll be affected by this change

File: invenio_app_ils/documents/jsonresolvers/document_stock.py [hunk 4]
```diff
@@ -32,9 +33,11 @@ def stock_resolver(document_pid):
         mediums = [
             bucket.key for bucket in search_response.aggregations.mediums.buckets
         ]
-        eitems_count = eitem_search.search_by_document_pid(document_pid).count()
-        if eitems_count > 0:
-            mediums.append("E-BOOK")
+        eitems = eitem_search.search_by_document_pid(document_pid)
+        for type in EItem.EITEM_TYPES:
+            type_count = eitems.filter("term", eitem_type=type).count()
```
Comment (ntarocco) [inline invenio_app_ils/documents/jsonresolvers/document_stock.py:38]: Did you check if `eitems` is immutable? When you use `.filter`, does it change the `eitems` object or simply return a new version with the filter applied?

File: invenio_app_ils/eitems/mappings/os-v1/eitems/eitem-v1.0.0.json [hunk 5]
```diff
@@ -61,6 +61,9 @@
       "document_pid": {
         "type": "keyword"
       },
+      "eitem_type": {
```
Comment (ntarocco) [inline invenio_app_ils/eitems/mappings/os-v1/eitems/eitem-v1.0.0.json]: do we need the prefix `eitem_`, given that we are already in the e-item?

File: invenio_app_ils/documents/jsonresolvers/document_stock.py (hunk 4 above)
Comment (sakshamarora1) [reply invenio_app_ils/documents/jsonresolvers/document_stock.py:38]: Yes, it returns a new version of the object with the filter applied on it, it doesn't apply it in-place

File: invenio_app_ils/eitems/mappings/os-v1/eitems/eitem-v1.0.0.json (hunk 5 above)
Comment (sakshamarora1) [reply invenio_app_ils/eitems/mappings/os-v1/eitems/eitem-v1.0.0.json]: My reasoning for the prefix `eitem_` was to be consistent with naming like `document_type` for documents. Wouldn't it be confusing with just type?

**Pull request merged**